### PR TITLE
Wire in the PurgePD capability check to the tx simulator

### LINKED ISSUE #3029: Introduce a new capability to enable processing purge operation

### PR DESCRIPTION
**Pull request by mbwhite**
Check the capability of PurgePD is enabled.

Resolves #3029

### CODE REVIEW

File: core/chaincode/handler.go [hunk 1]
```diff
@@ -553,6 +553,18 @@ func (h *Handler) checkMetadataCap(msg *pb.ChaincodeMessage) error {
...
 }
 
+func (h *Handler) checkPurgePrivateDataCap(msg *pb.ChaincodeMessage) error {
+	ac, exists := h.AppConfig.GetApplicationConfig(msg.ChannelId)
+	if !exists {
+		return errors.Errorf("application config does not exist for %s", msg.ChannelId)
+	}
+
+	if !ac.Capabilities().PurgePvtData() {
+		return errors.New("key level endorsement is not enabled, channel application capability of V1_3 or later is required")
```
Comment (denyeart) [inline core/chaincode/handler.go]: Update the error message to reference V2_5 application capability!

**Comment by mbwhite** [comment]
@manish-sethi @denyeart should be good for review now; took a while to track down which unittest was actually failing. Straightforward once that had been located.

**Review by denyeart — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -7,5 +7,25 @@ Please visit the
...
+A number of mock implementations of interfaces are used within the unit tests. For historical reasons there are two tools
+with the repo to generate these mocks. [mockery](https://github.com/vektra/mockery) and [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter)
+
+- look in the already created mock - the first line will indicate which tool it was created with
+- for counterfieter, run `go generate ./<dir>/...` in the <dir> where you want the mocks directory to be created
+
+- for mockery the command is `mockery --name ApplicationCapabilities --dir ~/github.com/hyperledger/fabric/common/channelconfig --filename application_capabilities.go`
+    - this will create the mock with the filename given in the mocks directory based of the cwd
+    - the `--name` and `--dir` indicate where the 'source' interface is to be mocked
+
```
Comment (denyeart) [inline CONTRIBUTING.md:29]: Thanks for adding the mock instructions!

**Comment by denyeart** [comment]
@Mergifyio backport release-2.5

**Pull request merged**